src/io/evented.cr

```diff
@@ -88,6 +88,23 @@ module IO::Evented
     end
   end
 
+  def evented_sendfile(remaining, errno_msg : String) : Nil
```

Comment:
I'm not sure about this method, for example `recvfrom` uses `evented_read` too. I think this is just telling us that `evented_write` is too specific and needs to be abstracted more to handle the `sendfile` case (i.e. move the slice handling into the callers).

I'm happy to do this refactor myself, though. For example, merge this then merge a refactor after.

Reply:
ok! sounds good